src/App.tsx

```diff
@@ -1,223 +1,197 @@
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [title, setTitle] = useState('');
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+  const [completedTodosId, setCompletedTodosId] = useState<number[]>([]);
+  const [toggle, setToggle] = useState<boolean>(true);
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
     }
+
+    getTodosFromServer(user.id);
   }, []);
 
+  const filterTodoBy = useMemo(() => getFilteredTodo(filterType, todos),
+    [todos, filterType]);
+
+  const newTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+    if (!title.trim() || !user) {
+      setErrorMessage(ErrorMessage.ErrorTitle);
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const postTodo = await createTodo(title, user.id);
+
+      setTodos((prevTodos) => [...prevTodos, postTodo]);
+    } catch {
+      setErrorMessage(ErrorMessage.NotAdd);
+    }
+
+    setIsAdding(false);
+    setTitle('');
+  }, [title, user]);
+
+  const removeTodo = useCallback(async (TodoId: number) => {
+    setSelectedIds([TodoId]);
+    try {
+      await deleteTodo(TodoId);
+
+      setTodos((prevTodos) => prevTodos.filter(({ id }) => id !== TodoId));
+    } catch {
+      setErrorMessage(ErrorMessage.NotDelete);
+    }
+  }, [todos, errorMessage]);
+
+  const completedTodos = useMemo(() => {
+    return todos.filter(todo => todo.completed);
+  }, [todos]);
+
+  const deleteCompletedTodos = useCallback(async () => {
+    setCompletedTodosId(completedTodos.map((todo => todo.id)));
+
+    await Promise.all(completedTodos.map(({ id }) => removeTodo(id)))
+      .then(() => setTodos((prevTodos) => prevTodos
+        .filter(({ completed }) => !completed)))
+      .catch(() => {
+        setErrorMessage(ErrorMessage.NotDelete);
+        setSelectedIds([]);
+      });
+
+    setCompletedTodosId([]);
+  }, [todos, selectedIds, errorMessage]);
+
+  const handleOnChange = useCallback(
+    async (updateId: number, data: Partial<Todo>) => {
+      setCompletedTodosId([updateId]);
+
+      try {
+        const changeStatus: Todo = await updateTodo(updateId, data);
+
+        setTodos(todos.map(todo => (
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos